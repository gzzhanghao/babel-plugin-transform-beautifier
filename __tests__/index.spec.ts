import * as babel from '@babel/core';
import deindent from 'deindent';

import beautifier from '../src';

function format(str: string) {
  return deindent(str).trim();
}

function t(name: string, input: string) {
  it(name, () => {
    const res = babel.transformSync(format(input), {
      filename: `${name}.js`,
      sourceType: 'script',
      plugins: [beautifier],
    });
    expect(res.code).toMatchSnapshot();
  });
}

describe('beautifier', () => {
  t('declaration', 'var a, b = 1');
  t('declaration full', 'var a = 1, b = 2, c = 3');
  t('declaration empty', 'var a, b, c');
  t('declaration in for', 'for (var a = 1, b = 2;;);');

  t('conditional', 'a ? b : c');
  t('conditional return', '() => { return a ? b : c }');
  t('conditional assign', 'a = b ? c : d');
  t('conditional deep', 'a(b ? c : d)');

  t('for stmt', 'for (;;);');
  t('for block', 'for (;;) {}');

  t('seq', 'a, b, c');
  t('seq return', '() => { return a, b, c }');
  t('seq deep', 'a((b, c, d))');

  t('logical and', 'a && b');
  t('logical or', 'a || b');
  t('logical deep', 'if (a && b);');

  t('unary true', '!0');
  t('unary false', '!1');
  t('unary undefined', 'void 0');
  t('unary void', 'void x()');
  t('unary void return', '() => { return void x() }');
  t('unary void deep', 'a(void b())');

  t('object method', '({ a: function() {} })');
  t('object arrow', '({ a: () => {} })');
  t('object value', '({ a: 123 })');

  t('if block', 'if (x) {}');
  t('if else block', 'if (x) {} else {}');
  t('if else logical', 'if (x); else a && b');
  t('if else conditional', 'if (x); else a ? b : c');
  t('if else conditional assign', 'if (x); else a = b ? c : d');

  t('template literal', '"a".concat(b).concat(c)');
  t('template literal with tail', '"a".concat(b).concat("c")');
});
