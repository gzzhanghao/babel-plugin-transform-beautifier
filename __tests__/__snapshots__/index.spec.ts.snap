// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`beautifier > conditional 1`] = `
"if (a) {
  b;
} else {
  c;
}"
`;

exports[`beautifier > conditional assign 1`] = `
"if (b) {
  a = c;
} else {
  a = d;
}"
`;

exports[`beautifier > conditional deep 1`] = `"a(b ? c : d);"`;

exports[`beautifier > conditional return 1`] = `
"() => {
  if (a) {
    return b;
  } else {
    return c;
  }
};"
`;

exports[`beautifier > declaration 1`] = `
"var a;
var b = 1;"
`;

exports[`beautifier > declaration empty 1`] = `"var a, b, c;"`;

exports[`beautifier > declaration full 1`] = `
"var a = 1;
var b = 2;
var c = 3;"
`;

exports[`beautifier > declaration in for 1`] = `
"var a = 1;
for (var b = 2;;) {
  ;
}"
`;

exports[`beautifier > for block 1`] = `"for (;;) {}"`;

exports[`beautifier > for stmt 1`] = `
"for (;;) {
  ;
}"
`;

exports[`beautifier > if block 1`] = `"if (x) {}"`;

exports[`beautifier > if else block 1`] = `"if (x) {} else {}"`;

exports[`beautifier > if else conditional 1`] = `
"if (x) {
  ;
} else if (a) {
  b;
} else {
  c;
}"
`;

exports[`beautifier > if else conditional assign 1`] = `
"if (x) {
  ;
} else if (b) {
  a = c;
} else {
  a = d;
}"
`;

exports[`beautifier > if else logical 1`] = `
"if (x) {
  ;
} else if (a) {
  b;
}"
`;

exports[`beautifier > logical and 1`] = `
"if (a) {
  b;
}"
`;

exports[`beautifier > logical deep 1`] = `
"if (a && b) {
  ;
}"
`;

exports[`beautifier > logical or 1`] = `
"if (!a) {
  b;
}"
`;

exports[`beautifier > object arrow 1`] = `
"({
  a: () => {}
});"
`;

exports[`beautifier > object method 1`] = `
"({
  a() {}
});"
`;

exports[`beautifier > object value 1`] = `
"({
  a: 123
});"
`;

exports[`beautifier > seq 1`] = `
"a;
b;
c;"
`;

exports[`beautifier > seq deep 1`] = `"a((b, c, d));"`;

exports[`beautifier > seq return 1`] = `
"() => {
  a;
  b;
  return c;
};"
`;

exports[`beautifier > template literal 1`] = `"\`a\${b}\${c}\`;"`;

exports[`beautifier > template literal with string 1`] = `""abc";"`;

exports[`beautifier > template literal with tail 1`] = `"\`a\${b}c\`;"`;

exports[`beautifier > unary false 1`] = `"false;"`;

exports[`beautifier > unary true 1`] = `"true;"`;

exports[`beautifier > unary undefined 1`] = `"undefined;"`;

exports[`beautifier > unary void 1`] = `
"x();
undefined;"
`;

exports[`beautifier > unary void deep 1`] = `"a(void b());"`;

exports[`beautifier > unary void return 1`] = `
"() => {
  x();
  return;
};"
`;
